src/main/java/uk/gov/companieshouse/company/profile/converter/CompanyProfileConverter.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.companieshouse.company.profile.converter;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.bson.Document;
+import org.bson.json.JsonWriterSettings;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.data.convert.ReadingConverter;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.company.Data;
+
+@ReadingConverter
+@Component
+public class CompanyProfileConverter implements Converter<Document, Data> {
+
+    private final ObjectMapper mapper;
+
+    public CompanyProfileConverter(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public Data convert(Document source) {
+        // Exclude properties with null values from being serialised
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        try {
+            // Use a custom converter for the ISO datetime stamps
+            JsonWriterSettings writerSettings = JsonWriterSettings
+                    .builder()
+                    .dateTimeConverter(new JsonDateTimeConverter())
```

Comment:
This will covert all time objects like Instant, LocalDate, LocalDateTime etc into one bson representation, not sure all fields in collection have same "date" type format

Reply:
In our case they do from what I can see.

Reply:
Good spot, all ISODates in mongo, and in the model all `Strings` just double checked - this is different to how it is for the Insolvency thankfully.